src/main/kotlin/org/kafkalytic/plugin/KafkaUtils.kt

```diff
@@ -254,15 +267,36 @@ fun produceGeneratedMessages(producer: KafkaProducer<ByteArray, ByteArray>, topi
     notify("Produced total $futures messages for topic $topic")
 }
 
-fun produceSingleMessage(producer: KafkaProducer<ByteArray, ByteArray>, topic: String, key: String, value: ByteArray) {
+fun produceSingleMessage(producer: KafkaProducer<ByteArray, ByteArray>, topic: String, key: String, value: ByteArray, header: String) {
     try {
-        producer.send(ProducerRecord<ByteArray, ByteArray>(topic, key.toByteArray(), value)).get()
+        val message = ProducerRecord<ByteArray, ByteArray>(topic, key.toByteArray(), value)
+        if(!header.isNullOrEmpty()){
+            //Custom WITHOUT HEADER
+            val header = createCustomHeader(header);
+            header.forEach {
+                message.headers().add(it);
+            }
+        }
+        producer.send(message).get()
         notify("Published $key")
     } catch (e: ExecutionException) {
         notify("Publish failed: $e")
     }
 }
 
+fun createCustomHeader(header: String, current: Int = -1 ): Headers {
+    var headerSuffix: String = if (current != -1) ("_$current") else "";
+    val Headers = RecordHeaders();
+    val listCustomHeader: List<String> = header?.split(";")
+    listCustomHeader.forEach {
+        val headerKey = header?.split(":")?.getOrNull(0)
+        val headerValue = header?.split(":")?.getOrNull(1)
+        Headers.add(RecordHeader(headerKey, (headerValue + headerSuffix).toByteArray(Charset.defaultCharset())))
+    }
+    return Headers;
+}
```

Comment:
fun createCustomHeader(header: String, current: Int = -1 ): Iterable<Header> {
    var headerSuffix: String = if (current != -1) "_$current" else "";
    return header.split(";").map {
        val (key, value) = it.split(":")
        RecordHeader(key, (value + headerSuffix).toByteArray(Charset.defaultCharset()))
    }
}